crates/ruma-client-api/src/room/get_summary.rs

```diff
@@ -0,0 +1,150 @@
+//! `GET /_matrix/client/v1/summary/{roomIdOrAlias}`
+//!
+//! Experimental API enabled with MSC3266.
+//!
+//! Returns a short description of a room.
+
+pub mod v1 {
+    //! `/v1/` ([MSC])
+    //!
+    //! [MSC]: https://github.com/deepbluev7/matrix-doc/blob/room-summaries/proposals/3266-room-summary.md
+
+    use ::serde::{Deserialize, Deserializer, Serialize};
+    use ruma_common::{
+        api::{request, response, Metadata},
+        metadata,
+        room::RoomType,
+        EventEncryptionAlgorithm, OwnedMxcUri, OwnedRoomAliasId, OwnedRoomId, OwnedRoomOrAliasId,
+        OwnedServerName, RoomVersionId,
+    };
+    use ruma_events::room::{join_rules::JoinRule, member::MembershipState};
+
+    const METADATA: Metadata = metadata! {
+        method: GET,
+        rate_limited: false,
+        authentication: AccessTokenOptional,
+        history: {
+            unstable => "/_matrix/client/unstable/im.nheko.summary/rooms/:room_id_or_alias/summary",
+        }
+    };
+
+    /// Request type for the `summary/room_id_or_alias` endpoint.
+    #[request(error = crate::Error)]
+    pub struct Request {
+        /// Alias or ID of the room to be summarized.
+        #[ruma_api(path)]
+        pub room_id_or_alias: OwnedRoomOrAliasId,
+
+        /// An optional list of servers the invited homeserver should attempt to peek at the room.
+        ///
+        /// If present, it must not be empty.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub via: Option<Vec<OwnedServerName>>,
```

Comment:
Should probably be a normal `Vec` and just default to empty, since I assume it is allowed to be empty since an alias is routable.